test/end2end_test.go

```diff
@@ -1037,7 +1051,10 @@ func (s) TestDetailedConnectionCloseErrorPropagatesToRpcError(t *testing.T) {
 	// the RPC has been started on it.
 	<-rpcStartedOnServer
 	ss.S.Stop()
-	if _, err := stream.Recv(); err == nil || (!strings.Contains(err.Error(), possibleConnResetMsg) && !strings.Contains(err.Error(), possibleEOFMsg)) {
+	// The precise behavior of this test is subject to raceyness around the timing
+	// of when TCP packets are sent from client to server, and when we tell the
+	// server to stop, so we need to account for both possible error messages.
+	if _, err := stream.Recv(); err == nil || !isConnClosedErr(err) {
```

Comment:
FWIW / for future reference, this will pass if `err == io.EOF` (due to the implementation of `isConnClosedErr`), which happens when streams are terminated successfully.  But, this is impossible in this test since the handler never returns nil, so I don't think anything needs to change here.

Reply:
I didnt get that. With the current the implementation, `isConnClosedErr(io.EOF)` = true. So wouldn't the check fail?

Reply:
Right, the check would fail and the test would pass.  But we don't want the test to pass if the RPC ends successfully, right?

Reply:
ah gotcha. 
i know you said nothing needs to change earlier. But for proper guardrails, should I remove the EOF check from `isConnClosedErr` and then add EOF check in the httpServer Start read loop?

Reply:
If you want to change anything, I'd _probably_ do it the other way around: adjust this call site since `stream.Recv` is probably a more "special" case:

```go
if _, err := stream.Recv(); err == io.EOF || !isConnClosedErr(err) {
```

Reply:
done